### ISSUE TITLE
UseSsl Ignored.

### ISSUE THREAD

**Issue by boggey79** [tanveery/recaptcha-net#2]
I am loading the Recaptcha (vr 1.7.0) into an MVC view using the following:

    @Html.Recaptcha(, Recaptcha.Web.RecaptchaTheme.Clean, Nothing, 0, True)

It seems the library is checking Request.IsSecureConnection() to determine whether or not the path should use SSL and overriding the UseSsl property set in the constructor. As I am behind a load balancer that is using SSL offloading, it would always return false if Request.IsSecureConnection() is called.

Can the value set in the constructor always take priority over what Request.IsSecureConnection()  returns?

Thanks for a great tool!

**Comment by tanveery** [comment]
Will do that in 1.8.

Thanks for using reCAPTCHA for .NET.

**Comment by boggey79** [comment]
Much appreciated...thank you!
I solved for now by just bypassing the @html helper and pasting in the raw script and js include.

**Comment by mabstrei** [comment]
This should do the trick ;-)

In RecaptchaHtmlHelper.cs

```
public RecaptchaHtmlHelper(string publicKey, RecaptchaTheme theme, string language, int tabIndex)
{
            RecaptchaHtmlHelper(string publicKey, RecaptchaTheme theme, string language, int tabIndex, HttpContext.Current.Request.IsSecureConnection)
}

public RecaptchaHtmlHelper(string publicKey, RecaptchaTheme theme, string language, int tabIndex, bool useSsl)
{
            this.PublicKey = RecaptchaKeyHelper.ParseKey(publicKey);

            if (String.IsNullOrEmpty(this.PublicKey))
            {
                throw new InvalidOperationException("Public key cannot be null or empty.");
            }

            this.Theme = theme;
            this.Language = language;
            this.TabIndex = tabIndex;

            this.UseSsl = useSsl;
}
```

In RecaptchaMvcExtensions.cs
```
public static IHtmlString Recaptcha(
            this HtmlHelper htmlHelper,
            string publicKey = "{recaptchaPublicKey}",
            RecaptchaTheme theme = RecaptchaTheme.Red,
            string language = null,
            int tabIndex = 0,
            bool useSsl = false)
{            
            RecaptchaHtmlHelper rHtmlHelper = new RecaptchaHtmlHelper(publicKey, theme, language, tabIndex, useSsl);

            HtmlTextWriter writer = new HtmlTextWriter(new StringWriter());
            writer.Write(rHtmlHelper.ToString());

            return htmlHelper.Raw(writer.InnerWriter.ToString());
}

**Comment by ajrowland** [comment]
Hi there.  Is this likely to be fixed/implemented any time soon?  Having the same issue.  Thanks, Andy.

**Comment by tanveery** [comment]
This issue is fixed. An new enum SslBehavior has been added and this enum is now the type of the UseSsl property / parameter. The UseSsl property is no longer a Boolean type. This fix is part of the 1.8.0.0 release and the same is updated on Nuget. The definition of the SslBehavior is as follows:

    public enum SslBehavior
    {
        /// <summary>
        /// Use SSL if the HttpContext.Current.Request.IsSecureConnection is True.
        /// </summary>
        SameAsRequestUrl = 0,
        /// <summary>
        /// Do not use SSL.
        /// </summary>
        DoNotUseSsl = 1,
        /// <summary>
        /// Always use SSL.
        /// </summary>
        AlwaysUseSsl = 2
    }
